### DISCUSSION TITLE
[Documentation] SQL (Relational) Databases, Partial Updates PATCH

### DISCUSSION THREAD

**Discussion by lucianoinso** [fastapi/fastapi#3997] [Questions] [answered]
I've had a bit of a trouble implementing the PATCH using databases, found a bunch of people who had problems too, after finally implementing it by myself found a more elegant answer in the SQLModel documentation.

URL of the tutorial I was following: https://fastapi.tiangolo.com/tutorial/sql-databases/
URL with the answer I was looking for which is not in the FastAPI documentation: https://sqlmodel.tiangolo.com/tutorial/fastapi/update/).

I started from the documentation that talks about partial updates (URL: https://fastapi.tiangolo.com/tutorial/body-updates/?h=partial#partial-updates-with-patch), but it uses a dictionary as a database, came into difficulties translating the PATCHing function into something that works with databases (e.g: spent a lot of time checking why my model didn't have the copy method).

It would be really helpful if the SQL Database tutorial in the FastAPI documentation had all the functions for implementing the CRUD (PATCH, UPDATE and DELETE are missing), spent a bunch of time trying to figure things out of why it didn't work, thought it was something to do with Pydantic models and SQLAlchemy models, which I previously thought was one of the greatest strength of FastAPI, the natural way it uses and connects both.

So the change I'm suggesting is: add the SQLmodel documentation about how to implement a partial update using PATCH to the SQL Databases tutorial on FastAPI documentation (maybe add DELETE and UPDATE too).

Started using FastAPI a couple of days ago, by so far loved the framework, although hated it a bit today.

**Comment by YuriiMotov** [comment, marked as answer]
Current version points to SQLMOdel docs for further reading:

> This is a very simple and short tutorial, if you want to learn about databases in general, about SQL, or more advanced features, go to the [SQLModel docs](https://sqlmodel.tiangolo.com/).
